packages/apollo-voyager-metrics/src/index.ts

```diff
@@ -33,64 +31,138 @@ const serverResponseMetric = new Prometheus.Histogram({
   labelNames: ['request_type', 'error']
 })
 
-export function getMetrics (req: IncomingMessage, res: Response) {
-  res.set('Content-Type', Prometheus.register.contentType)
-  res.end(Prometheus.register.metrics())
+interface ResolverObject {
+  [key: string]: IFieldResolver<any, any>
+}
 
-  resolverTimingMetric.reset()
-  resolverRequestsMetric.reset()
-  serverResponseMetric.reset()
+export function enableDefaultMetricsColleciton () {
+  if (!promMetricsEnabled) {
+    Prometheus.collectDefaultMetrics()
+    promMetricsEnabled = true
+  }
 }
 
-export function responseLoggingMetric (req: IncomingMessage, res: ResponseWithVoyagerMetrics, next: NextFunction) {
-  const requestMethod = req.method as string
+export function wrapResolversForMetrics (resolvers: {[key: string]: ResolverObject}): {[key: string]: ResolverObject} {
+  const output: {[key: string]: ResolverObject} = {}
+  for (const typeKey in resolvers) {
+    if (!resolvers.hasOwnProperty(typeKey)) {
+      continue
+    }
+    const fieldResolversForType = resolvers[typeKey]
+    output[typeKey] = {}
+    for (const fieldKey in fieldResolversForType) {
+      if (!fieldResolversForType.hasOwnProperty(fieldKey)) {
+        continue
+      }
+      const resolverForField = fieldResolversForType[fieldKey]
+      output[typeKey][fieldKey] = wrapSingleResolverForMetrics(resolverForField)
+    }
+  }
 
-  res.requestStart = Date.now()
+  return output
+}
 
-  res.on('finish', onResFinished)
-  res.on('error', onResFinished)
+export function applyResponseLoggingMetricsMiddleware (app: Application) {
+  app.use(responseLoggingMetric as (req: IncomingMessage, res: Response, next: NextFunction) => void)
```

Comment:
Eventually we might be able to turn this into an interface and support more http frameworks than just express so +1 for this.